.github/workflows/release-validation.yaml

```
@@ -0,0 +1,24 @@
+name: Release Validation
+
+on:
+  pull_request:
+    types:
+    - opened
+    branches:
+      - 'release-v[0-9]+.[0-9]+'
```

If this is meant for validating the RC branch, maybe the following will target better the event:

```suggestion
      - 'release-v[0-9]+.[0-9]+-?**'
```

---

I am not a fan of using regex for any of this. I know some is a skill issue on my behalf, but that is why I am not going to change this. What you have suggested is not valid on https://regex101.com/ and as someone that does not have a depth understand of regex that is a tool I would use to validated what your suggesting. If I can't validate it now, a future contributor may also have the same issue.

---

The first one would still capture the RC right?
maybe something like? 
```
release-(v[0-9]+.[0-9])+(-[\w]+)? 
```
could be a minor improvement as it would capture the version and the rc for potential use? (by the way my regex foo is also weak)

---

Sadly GH actions don't support full Regex, so the suggestion is considering GH limitations. We are currently using it in some workflows, here the PR reckoning the behaviour: https://github.com/Kuadrant/limitador-operator/pull/181 and https://github.com/Kuadrant/kuadrant-operator/pull/1162

---

More food for thought: check out my comment regarding this: https://github.com/Kuadrant/limitador-operator/pull/181#pullrequestreview-2596181105